src/primval.rs

```diff
@@ -72,11 +72,11 @@ pub fn binary_op<'tcx>(bin_op: mir::BinOp, left: PrimVal, right: PrimVal) -> Eva
                 Rem    => $v($l % $r),
 
                 // invalid float ops
-                BitXor => unreachable!(),
-                BitAnd => unreachable!(),
-                BitOr  => unreachable!(),
-                Shl => unreachable!(),
-                Shr => unreachable!(),
+                BitXor => bug!("`^` is not a valid operation on floats"),
+                BitAnd => bug!("`&` is not a valid operation on floats"),
+                BitOr  => bug!("`|` is not a valid operation on floats"),
+                Shl => bug!("`<<` is not a valid operation on floats"),
+                Shr => bug!("`>>` is not a valid operation on floats"),
```

Comment:
Since there's no existing method mapping `BinOp`s to their symbols, I'll just merge this as-is.